File: pset6/caesar.py
import sys
import cs50

MOD = 26

def is_number(number):
    try:
        int(number)
        return True
    except ValueError:
        return False

def get_base_char(letter):
    if letter.isupper():
        return 'A'
    return 'a'

def encode_char(letter, shift):
    if (letter.isalpha()):
        return chr(ord(get_base_char(letter)) + (ord(letter) - ord(get_base_char(letter)) + int(shift)) % MOD)
    return letter

def cipher_encode(text, shift):
    encoded = "";
    for letter in text:
        encoded += encode_char(letter, shift)
    return encoded

def main(argc, argv):
    if argc != 2:
        print("Usage: caesar.py shift")
        quit()
    if not is_number(argv[1]):
        print("Second param should be a number")
        quit()
    print("plaintext: ", end="");
    text = cs50.get_string()
    shift = argv[1]
    print("ciphertext: %s" % cipher_encode(text, shift))


main(len(sys.argv), sys.argv)

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

#define UPPERCASE 'A'
#define LOWERCASE 'a'
#define ALPHA_LENGTH 26

char get_base_char(char c)
{
    return isupper(c) ? UPPERCASE : LOWERCASE;
}

char proccess_alpha(char c, int key)
{
    char base = get_base_char(c);
    return base + (c - base + key) % ALPHA_LENGTH;
}

string encode(string plain, int key)
{
    for (int i = 0; i < strlen(plain); i++)
    {
        if (isalpha(plain[i]))
        {
            plain[i] = proccess_alpha(plain[i], key);
        }
    }

    return plain;
}

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    int key = atoi(argv[1]);
    printf("plaintext: ");
    string plaintext = get_string();
    printf("ciphertext: %s\n", encode(plaintext, key));

    return 0;
}